src/main/java/ch/njol/skript/entity/BoatChestData.java

```diff
@@ -59,23 +74,23 @@ protected boolean init(Literal<?>[] exprs, int matchedPattern, SkriptParser.Pars
 	}
 
 	@Override
-	protected boolean init(@Nullable Class<? extends ChestBoat> c, @Nullable ChestBoat e) {
-		if (e != null)
-			matchedPattern = 2 + e.getWoodType().ordinal();
+	protected boolean init(@Nullable Class<? extends ChestBoat> c, @Nullable ChestBoat entity) {
+		if (entity != null)
+			matchedPattern = 2 + entity.getBoatType().ordinal();
 		return true;
 	}
 
 	@Override
 	public void set(ChestBoat entity) {
 		if (matchedPattern == 1) // If the type is 'any boat'.
-			matchedPattern += new Random().nextInt(TreeSpecies.values().length); // It will spawn a random boat type in case is 'any boat'.
+			matchedPattern += new Random().nextInt(Boat.Type.values().length); // It will spawn a random boat type in case is 'any boat'.
 		if (matchedPattern > 1) // 0 and 1 are excluded
```

Comment:
```suggestion
		else if (matchedPattern > 1) // 0 and 1 are excluded
```

Reply:
no?  if I do that, then if you use the any syntax, then it wont set the type.